src/components/widgets/gcode-preview/GcodePreview.vue

```diff
@@ -385,7 +389,15 @@ export default class GcodePreview extends Mixins(StateMixin) {
       }
     }
 
-    if (stepperX === undefined || stepperY === undefined) {
+    if (stepperX === undefined || stepperY === undefined || this.autoZoom) {
+      if (this.autoZoom) {
+        const padding = Math.min(bounds.x.max - bounds.x.min, bounds.y.max - bounds.y.min) * 0.05
+        bounds.x.min -= padding
+        bounds.x.max += padding
+        bounds.y.min -= padding
+        bounds.y.max += padding
```

Comment:
This is altering the `bounds` object directly!!

If I keep toggling the "magnifying" icon, it keeps changing the zoomed view as the bounds are also changing!